### PR TITLE
[ALFMOB-448] Modern Design Rollout — app shell & bottom bar

### PR DESCRIPTION
**Pull request by hoangnhatdrk**
Ticket: [ALFMOB-448](https://mindera.atlassian.net/browse/ALFMOB-448) · Epic: [ALFMOB-428](https://mindera.atlassian.net/browse/ALFMOB-428)
Figma: [Bottom Navigation](https://www.figma.com/design/axx7Bz1fpQurtU6DHwVaJX/Alfie---Designs--Mobile-?node-id=1-10084) — Android instance `671:77976`

Restyles the global bottom navigation bar to the modern design. Assembled entirely from existing `designsystem` components and `LocalTheme.current` tokens — **no new components, tokens, or icons were needed**.

## Bottom bar (`designsystem/component/bottombar/BottomBar.kt`)

- Colours move from raw primitives to the semantic layer: `surface.backgroundPrimary`, `border.soft`, `content.contentPrimary` / `contentTerciary`.
- **Removed the drop shadow and the 2 dp animated sliding indicator.** Figma has neither — selection is conveyed by colour alone. This takes `BottomBarIndicator`, `drawIndicator` and the `onGloballyPositioned` position-tracking plumbing with it.
- Fixed `63.dp` bar height and `86.dp` item minimum replaced by Figma's intrinsic box: padding 8 top / 16 bottom, items at `weight(1f)`, 8 gutter, 32 dp icon row, 4 dp icon-to-label gap. Resolves to exactly **76 dp**, matching the node.
- Label style `body.small` → `label.smallBold` (same 12/16 metrics, W500).
- Top border drawn as a rect of `primitive.border.weightDefault` instead of a centred `drawLine` at y=0, which clipped half the stroke and rendered ~0.5 dp. Pre-existing bug.
- Added `selected` semantics for TalkBack; `combinedClickable` kept verbatim so double-tap behaviour is unchanged.
- Previews rebuilt as the full Figma bar, plus a badged state.

## Tabs

- **Store icon** `LegacyStore` → `Menu` (`ic_menu` matches the Figma glyph pixel-for-pixel).
- **Tab label** "Shop" → "Store". Resource key stays `bottom_bar_shop` — it names the destination.
- **Account tab added** (5th in Figma) using the existing `AccountScreenDestination`.
- **Wishlist is now unconditional** — it sat behind the `Show Wishlist On Bottom Bar` debug toggle, which defaults off, so the shipped bar had 4 tabs while Figma shows 5 with no variant for its absence. The toggle and its full chain are removed: `MainActivityViewModel` deleted (the toggle was its only job), `wishlistToggleEnabled` unthreaded from `AppNavigation`, and the registry entry + string dropped.

## Behavioural changes (all agreed before implementation)

The ticket lists destination-set changes as out of scope; these were explicitly approved because Figma disagreed with the shipped app:

1. **Account added** to the destination set.
2. **`AccountScreen` switches `hideBottomBar()` → `showBottomBar()`** — without it the new tab is a navigation dead-end, since every other tab screen shows the bar.
3. **Wishlist no longer toggle-gated.**

## Verification

Compile ✅ · `detekt` ✅ **0 smells** · unit tests ✅ (`:app` and `:designsystem` are `NO-SOURCE` — neither has tests).

Driven on a **clean install** (Pixel 10 Pro AVD, prior install removed so no stored toggle rows could mask the result):

- All 5 tabs render, **no truncation or overflow at 360 dp** — the density the `weight(1f)` change was made for.
- Flat bar, 1 dp top rule, no shadow, no indicator; selected `#111111` vs unselected `#767676` on both icon and label.
- Account and Wishlist both navigate with the bar visible and selection tracking correct.
- The now-empty debug Feature Toggle screen renders blank without crashing.

Visual match against the Figma render confirmed.

## Notes for review

- **Glass Effect not implemented** — Figma applies a backdrop blur, but Compose has no portable equivalent and the bar's background is opaque white, so it's a visual no-op.
- **Figma's Bag icon is near-black while its own label is grey**, unlike every other unselected tab. Read as a detached-instance artefact and implemented as icon-tint-follows-label. Worth a design confirmation.
- **The Shop screen header still reads "Shop"** while the tab reads "Store" — deliberately left for the Shop screen's own rollout ticket.
- **Account's top bar still shows a back arrow** now that it's a top-level tab, and `AccountTopBarAction` is now a second entry point. Both belong to the top-bar rollout, which carries no Figma node in this ticket.
- **The debug Feature Toggle screen is now blank** — wishlist was its only entry. The generic infrastructure is intact, so the next toggle is a one-line addition.
- No screenshot-test infra exists in the repo (grepped, not assumed), so any "regenerate screenshot baselines" AC wording is N/A.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01URRUZj81wAWwwueS7qPrUT

[ALFMOB-448]: https://mindera.atlassian.net/browse/ALFMOB-448?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ALFMOB-428]: https://mindera.atlassian.net/browse/ALFMOB-428?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt [hunk 1]
```diff
@@ -153,143 +126,109 @@ private fun BottomBarItem(
...
                 indication = ripple(bounded = false)
             )
-            .defaultMinSize(minHeight = BOTTOM_BAR_HEIGHT)
-            .widthIn(min = 86.dp)
-            .onGloballyPositioned { coordinates ->
-                val bounds = coordinates.boundsInWindow()
-                onItemPosition(bounds.topLeft, bounds.bottomRight)
-            },
-        contentAlignment = Alignment.Center
+            .semantics { selected = state.isSelected },
```
Comment (khoinguyen-mindera) [inline designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt]: **Important — a11y: label doesn't merge into the Tab node.** `combinedClickable(role = Role.Tab)` + `.semantics { selected }` are on this `Column`, but nothing sets `mergeDescendants` and the icon is `contentDescription = null`. So the label `Text` stays a separate semantics node — TalkBack focuses the Tab node (role + selected, but **no name**) and the label as two separate stops, rather than announcing "Home, tab, selected" as one unit. Since TalkBack support is a stated goal of this PR, this is the one gap in it.

```kotlin
.semantics(mergeDescendants = true) { selected = state.isSelected }
```

File: designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt [hunk 2]
```diff
@@ -153,143 +126,109 @@ private fun BottomBarItem(
...
-                color = color
-            )
         }
+        // D7: `label/small-bold` (Roboto Medium 12/16) replaces `body.small` (same metrics, Normal).
+        Text(
+            text = label,
+            style = theme.typography.label.smallBold,
+            color = color,
+            textAlign = TextAlign.Center,
+            maxLines = 1,
```
Comment (khoinguyen-mindera) [inline designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt:182]: **Important — label can hard-clip.** `maxLines = 1` + `fillMaxWidth` + `TextAlign.Center` with no `overflow` defaults to `TextOverflow.Clip`. Each of 5 equal-weight tabs is ~65 dp at 360 dp; at font scale ≥1.3× (or narrower devices) "Wishlist"/"Account" clip mid-glyph with no ellipsis. The old code had no `maxLines`, so this risk is introduced here. Manual verification at 360 dp / default scale doesn't exercise it.

```kotlin
overflow = TextOverflow.Ellipsis,
```
and consider adding a `@Preview(fontScale = 1.5f)`.

**Review by khoinguyen-mindera — COMMENTED** [review]
Code review — 3 important points (2 inline below). No blockers; all fix-or-ticket.

**Important — orphaned `GetFeatureToggleUseCase` slice** (not inline: the file isn't in this diff)

Deleting `MainActivityViewModel` removed the only production consumer of `GetFeatureToggleUseCase`. It — and its whole tail — is now reachable only from tests: `getFeatureTogglesByNameAsFlow` on `FeatureToggleRepository` / `FeatureToggleRepositoryImpl:27` / `FeatureToggleDao:17` has no other caller. A test with no production caller isn't a seam earning its keep. Either remove the slice (use-case → repository method → DAO) in this PR, or note why it's retained for a near-term feature.

_Note: I could not run `./gradlew testDebugUnitTest` locally (no JDK on the review machine), so tests are CI-reported, not reviewer-verified — please confirm CI is green before merge._

File: designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt (hunk 2 above)
Comment (khoinguyen-mindera) [inline designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt:182]: **Accessibility (important):** this label has `maxLines = 1` + `fillMaxWidth()` but no `overflow`, so it defaults to `TextOverflow.Clip`. With 5 weighted tabs on 360 dp each item is ~65 dp wide; at a large accessibility font scale the longer labels ("Wishlist", "Account") exceed that and get hard-clipped mid-glyph with no ellipsis cue. The "no truncation at 360 dp" check was at default font scale, so this path wasn't exercised.

```suggestion
            maxLines = 1,
            overflow = TextOverflow.Ellipsis,
```
(add import `androidx.compose.ui.text.style.TextOverflow`)

**Review by khoinguyen-mindera — COMMENTED** [review]
Two review notes from an automated multi-lens code review.

**Orphaned use case (nit / follow-up — not in this diff):** `GetFeatureToggleUseCase` (singular, `domain/.../usecase/featuretoggle/GetFeatureToggleUseCase.kt`) was consumed only by `MainActivityViewModel`, which this PR deletes. Post-merge it and its test are referenced only by their own test; `GetAllFeatureToggleUseCase` remains the live path. Not deleting here since it's outside this diff — worth a follow-up ticket to remove it if no upcoming feature needs single-toggle lookup.

(Inline note below on the tab label overflow.)

**Review by Prathimaboggu — COMMENTED** [review]
@hoangnhatdrk Strong PR — I checked every number against Figma dev mode and found no mismatches, which is unusual. Nothing new from me is blocking.

**The main thing: all four points already raised are still unaddressed.** Both commits (`62f99dc` 07-26T01:53, `6a3250c` 07-26T09:02) predate every review (Copilot 07-26T09:18, @khoinguyen-mindera 07-27T06:07/06:14), so none of these have been picked up yet. I agree with all four and won't re-argue them:

| Raised by | Point |
|---|---|
| Copilot | Container-level 8/16 padding shrinks each tab's hit target — move it onto `BottomBarItem` before `.combinedClickable` |
| @khoinguyen-mindera | Label doesn't merge into the Tab node — needs `semantics(mergeDescendants = true)`; TalkBack currently gets a nameless Tab plus a separate label stop |
| @khoinguyen-mindera | `maxLines = 1` with no `overflow` → hard clip at large font scale; add `TextOverflow.Ellipsis` + a `fontScale` preview |
| @khoinguyen-mindera | Orphaned `GetFeatureToggleUseCase` slice now that `MainActivityViewModel` is gone |

The `mergeDescendants` one is the most important — TalkBack support is a stated goal of this PR, and that's the one gap in it.

### Figma verification — every value matches

### The upgrade path you sidestepped is safe

### Minor

- `ACCOUNT_TAB` is unused — nothing in the repo consumes any tab constant. Fine to land alongside the tab, just noting it buys nothing today.
- `FeatureToggleUIFactory` still injects `dispatcher` and wraps a bare `emptyList()` in `withContext(dispatcher.default())`.
- `bottom_bar_shop` → "Store" is safe: `BottomBarDestination.kt:72` is its only consumer, so keeping the key is contained.

### Verification gap

`testDebugUnitTest` doesn't cover instrumented UI, and there are no bottom-bar tests in any source set — so "all 5 tabs render, no truncation or overflow at 360 dp" rests on one manual pass at one density and default font scale. That's exactly the hole @khoinguyen-mindera's clipping finding sits in: it only appears at font scale ≥1.3×, which that check didn't exercise. The `@Preview(fontScale = 1.5f)` they suggested is the cheapest way to close it.

Same caveat they gave: I have no Android SDK on this machine, so the compile/detekt/test results in the description are author-reported and unverified by me.

**Comment by hoangnhatdrk** [comment]
Thanks @khoinguyen-mindera @Prathimaboggu — all four points addressed in `beeb76d`. CI green.

I had an emulator available, so everything below is verified on device rather than read from source. That turned up one correction and one finding that doesn't hold.

### 1. Tab hit target — fixed, but not as specified

The suggestion (Copilot's, seconded by @Prathimaboggu) was to move the padding onto `BottomBarItem` **before** `.combinedClickable`. That would not have worked — `Modifier.padding().clickable()` leaves the padding *outside* the touch node, which is the bug. Modifiers apply outside-in, so the clickable has to come first:

```kotlin
.combinedClickable(...)
.padding(top = theme.spacing.spacing8, bottom = theme.spacing.spacing16)
```

Padding removed from the container `Row`. Bar height is unchanged at 76dp and the `drawBehind` top border still draws at the Row's edge.

Verified: the clickable node's bounds are now the full cell (222px ≈ 76dp at this density), and a tap 4px below the cell's top edge — previously dead space — selects the tab.

### 2. Label clipping — confirmed real, fixed

`overflow = TextOverflow.Ellipsis` added, plus a `@Preview(fontScale = 1.5f)`.

Your reasoning was exactly right, including why the original manual pass missed it. At the emulator's native ~426dp the labels fit at 1.5× and nothing reproduces. Forcing a 360dp/1.5× config, "Account" clips — and now renders `Accou…` instead of a hard cut mid-glyph.

### 3. `mergeDescendants` — this one doesn't hold

You flagged this as "the one gap in this PR" and @Prathimaboggu ranked it the most important of the four. I don't think it changes anything, and I'd rather say so than quietly ship a no-op.

`combinedClickable` already applies `semantics(mergeDescendants = true)` internally, so the Column was a merging node before this PR. I A/B tested it — built with and without the change and dumped the accessibility tree each time. **The two trees are identical.** In both, each tab is a single `focusable=true` node spanning the full cell with `selected` set, and the label `TextView` is a non-focusable child, i.e. not a separate stop.

Enabling TalkBack confirms it: the focus rectangle wraps icon + label as one unit, on the build *without* the change.

So TalkBack was already announcing the tab as one stop — there was no nameless Tab node and no second stop for the label.

I've kept the explicit `mergeDescendants = true` anyway, because the single-stop announcement otherwise rests on a `combinedClickable` implementation detail rather than anything contractual. But the code comment and commit message now both say plainly that it's a no-op today. Shout if you'd rather I drop it entirely.

### 4. Orphaned `GetFeatureToggleUseCase` — deleted

Confirmed your call: zero production callers once `MainActivityViewModel` goes, only its own test. Removed the use case, its test, the `FeatureToggleRepository` declaration and impl override, and the `FeatureToggleDao` query. `GetAllFeatureToggleUseCase` and the rest of the chain stay live via `FeatureToggleViewModel`.

### Minor

- **`FeatureToggleUIFactory`** — dropped the `DispatcherProvider` and the `withContext` around a constant `emptyList()`, as @Prathimaboggu noted.
- **`ACCOUNT_TAB`** — left as-is. It is unused, but so are the other four tab constants: nothing in the repo consumes any of them (there are no tests in `:app` or `:designsystem` at all). Removing only the newest would leave the set inconsistent. Better handled whenever tab tests actually land.

### Verification

`./gradlew compileDebugKotlin detekt` across all modules, plus `testDebugUnitTest` on `:domain`, `:data` and `:debug:operational`. On device: all five tabs at default and 1.5× scale, the padding-region tap, and the TalkBack focus check above.

Worth flagging for whoever picks up tab tests — there is no Compose UI test infrastructure in this repo at all: no `ui-test-junit4` in the version catalog, no `createComposeRule` anywhere, no screenshot tooling. The `fontScale` preview is the cheap mitigation; real coverage needs its own ticket.

File: designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt [hunk 3]
```diff
@@ -105,44 +82,40 @@ fun BottomBar(
...
-                verticalAlignment = Alignment.CenterVertically,
-                content = content
-            )
-            BottomBarIndicator(
-                startPosition = selectedStartPosition,
-                endPosition = selectedEndPosition,
-                modifier = Modifier.align(Alignment.TopCenter)
-            )
-        }
+                .selectableGroup(),
```
[the remark this answers is not in the document]
Comment (hoangnhatdrk) [reply designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt]: Fixed in `beeb76d`, though with the modifier order reversed from the suggestion: `.padding()` *before* `.combinedClickable()` leaves the padding outside the touch node, which is the bug. It has to be `.combinedClickable(...).padding(...)`, since modifiers apply outside-in.

Verified on device — the clickable node is now the full 76dp cell, and a tap 4px below its top edge selects the tab. Details in the [main comment](https://github.com/Mindera/Alfie-Android/pull/35#issuecomment-5117964869).

File: designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt (hunk 1 above)
Comment (hoangnhatdrk) [reply designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt]: I looked hard at this one and I do not think it holds — `combinedClickable` already applies `semantics(mergeDescendants = true)` internally, so the Column was a merging node before this PR.

A/B tested it: built with and without, dumped the accessibility tree each time, and the trees are identical. Each tab is a single `focusable=true` node spanning the cell with `selected` set; the label `TextView` is a non-focusable child, not a separate stop. TalkBack confirms it — the focus rect wraps icon + label as one unit on the build *without* the change.

I have kept the explicit `mergeDescendants = true` so the behaviour does not rest on a `combinedClickable` implementation detail, but the comment and commit message now both say it is a no-op today. Happy to drop it if you would rather. Full evidence in the [main comment](https://github.com/Mindera/Alfie-Android/pull/35#issuecomment-5117964869).

File: designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt (hunk 2 above)
Comment (hoangnhatdrk) [reply designsystem/src/main/java/com/mindera/alfie/designsystem/component/bottombar/BottomBar.kt:182]: Applied in `beeb76d`, plus a `@Preview(fontScale = 1.5f)`.

Confirmed real, and your note about why the manual pass missed it was exactly right: at the emulator native ~426dp everything fits at 1.5x. Forcing 360dp/1.5x, "Account" now renders `Accou…` where it previously cut mid-glyph.

**Review by khoinguyen-mindera — APPROVED** [review]
Approving — re-reviewed at head `18ee4b5`.

All four points raised across the reviews are resolved and verified against the code:
- **Tab hit target** — padding moved onto the item *after* `.combinedClickable` (inside the touch node); 76 dp cell now fully tappable.
- **Label clipping** — `overflow = TextOverflow.Ellipsis` + a `@Preview(fontScale = 1.5f)` guard.
- **TalkBack `mergeDescendants`** — added explicitly and honestly documented as a no-op today; harmless and defensible.
- **Orphaned `GetFeatureToggleUseCase`** — use case, its test, repo interface decl + impl override, and DAO query all removed; zero dangling references remain and the `GetAllFeatureToggleUseCase` chain stays live.

No new issues introduced by `beeb76d`; `FeatureToggleUIFactory` correctly simplified and its test updated (`coEvery`→`every`). CI green on this head.

Caveat: build/tests verified via CI + author's device run, not independently reproduced in my review environment (no local JDK). No blockers — good to merge.
